DFC.FindACourseClient/FindACourseClient.cs

```diff
@@ -0,0 +1,68 @@
+﻿using DFC.FindACourseClient.Contracts;
+using DFC.FindACourseClient.Models.APIRequests;
+using DFC.FindACourseClient.Models.APIResponses.CourseGet;
+using DFC.FindACourseClient.Models.APIResponses.CourseSearch;
+using DFC.FindACourseClient.Models.Configuration;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System;
+using System.Net.Http;
+using System.Net.Http.Formatting;
+using System.Threading.Tasks;
+
+namespace DFC.FindACourseClient
+{
+    public class FindACourseClient : IFindACourseClient
+    {
+        private readonly ILogger logger;
+        private readonly CourseSearchClientSettings courseSearchClientSettings;
+        private readonly Guid correlationId;
+        private readonly IAuditService auditService;
+        private readonly HttpClient httpClient;
+
+        public FindACourseClient(HttpClient httpClient, CourseSearchClientSettings courseSearchClientSettings, IAuditService auditService, ILogger logger = null)
+        {
+            correlationId = Guid.NewGuid();
+            this.logger = logger;
+            this.auditService = auditService;
+            this.courseSearchClientSettings = courseSearchClientSettings ?? throw new ArgumentException(nameof(courseSearchClientSettings));
+
+            this.httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            this.httpClient.Timeout = TimeSpan.FromSeconds(courseSearchClientSettings.CourseSearchSvcSettings.RequestTimeOutSeconds);
+            this.httpClient.DefaultRequestHeaders.Add("Ocp-Apim-Subscription-Key", courseSearchClientSettings.CourseSearchSvcSettings.ApiKey);
+        }
+
+        public async Task<CourseRunDetailResponse> CourseGetAsync(CourseGetRequest courseGetRequest)
+        {
+            var url = $"{courseSearchClientSettings.CourseSearchSvcSettings.ServiceEndpoint}courserundetail?CourseId={courseGetRequest.CourseId}&CourseRunId={courseGetRequest.RunId}";
+            var response = await httpClient.GetAsync(url).ConfigureAwait(false);
+            var responseContent = await (response?.Content?.ReadAsStringAsync()).ConfigureAwait(false);
+
+            await auditService.CreateAudit(courseGetRequest, responseContent, correlationId).ConfigureAwait(false);
```

Comment:
This should be in finally block so that it gets triggered no matter the response

Reply:
Amended as suggested